vnext/Chakra/ChakraHelpers.cpp

```diff
@@ -60,163 +60,43 @@ struct FileVersionInfoResource {
-  }
-
-  bool operator!=(const ChakraVersionInfo &rhs) const noexcept {
-    return !(*this == rhs);
-  }
-
- private:
-  uint32_t m_fileVersionMS;
-  uint32_t m_fileVersionLS;
-  uint32_t m_productVersionMS;
-  uint32_t m_productVersionLS;
-};
-
-class BytecodePrefix {
- public:
-  static std::pair<bool, BytecodePrefix> getBytecodePrefix(JsValueRef scriptFileName, uint64_t bundleVersion) noexcept {
-    std::pair<bool, BytecodePrefix> result{false, BytecodePrefix{bundleVersion}};
-    result.first = result.second.m_chakraVersionInfo.initialize();
-    return result;
-  }
-
-  bool operator==(const BytecodePrefix &rhs) const noexcept {
-    return (m_bytecodeFileFormatVersion == rhs.m_bytecodeFileFormatVersion) &&
-        (m_bundleVersion == rhs.m_bundleVersion) && (m_chakraVersionInfo == rhs.m_chakraVersionInfo);
-  }
-
-  bool operator!=(const BytecodePrefix &rhs) const noexcept {
-    return !(*this == rhs);
-  }
-
- private:
-  uint64_t m_bytecodeFileFormatVersion;
-  uint64_t m_bundleVersion;
-  ChakraVersionInfo m_chakraVersionInfo;
-
-  static constexpr uint32_t s_bytecodeFileFormatVersion = 2;
-
-  BytecodePrefix(uint64_t bundleVersion) noexcept
-      : m_bytecodeFileFormatVersion{s_bytecodeFileFormatVersion},
-        m_bundleVersion{bundleVersion},
-        m_chakraVersionInfo{} {}
-};
-
-void serializeBytecodeToFileCore(
+void persistBytecodeCore(
     const std::shared_ptr<const JSBigString> &script,
-    const BytecodePrefix &bytecodePrefix,
-    const std::string &bytecodeFileName) {
-  FILE *bytecodeFilePtr;
-  // bytecode is a binary representation, so we need to pass in the "b" flag to
-  // fopen_s
-  if (_wfopen_s(&bytecodeFilePtr, Microsoft::Common::Unicode::Utf8ToUtf16(bytecodeFileName).c_str(), L"wb")) {
-    return;
-  }
-  std::unique_ptr<FILE, decltype(&fclose)> bytecodeFilePtrWrapper(bytecodeFilePtr, fclose);
-
+    const std::string scriptUrl,
+    ChakraBytecodeStore &bytecodeStore) {
   const std::wstring scriptUTF16 = Microsoft::Common::Unicode::Utf8ToUtf16(script->c_str(), script->size());
 
   unsigned int bytecodeSize = 0;
   if (JsSerializeScript(scriptUTF16.c_str(), nullptr, &bytecodeSize) != JsNoError) {
     return;
   }
 
-  std::unique_ptr<uint8_t[]> bytecode(std::make_unique<uint8_t[]>(bytecodeSize));
-  if (JsSerializeScript(scriptUTF16.c_str(), bytecode.get(), &bytecodeSize) != JsNoError) {
-    return;
-  }
-
-  constexpr size_t bytecodePrefixSize = sizeof(bytecodePrefix);
-  uint8_t zeroArray[sizeof(bytecodePrefix)]{};
-
-  if (fwrite(zeroArray, bytecodePrefixSize, bytecodeFilePtrWrapper.get()) != bytecodePrefixSize ||
-      fwrite(bytecode.get(), bytecodeSize, bytecodeFilePtrWrapper.get()) != bytecodeSize ||
-      fflush(bytecodeFilePtrWrapper.get())) {
-    return;
-  }
-
-  fseek(bytecodeFilePtrWrapper.get(), 0, SEEK_SET);
-  if (!fwrite(bytecodePrefix, bytecodeFilePtrWrapper.get()) || fflush(bytecodeFilePtrWrapper.get())) {
+  JSBigBufferString bytecode(bytecodeSize);
```

Comment:
Don't love this since JSBigBufferStrings claims it's ASCII, which is a lie here. It ends up not doing anything for our implentation, but it would probably be better for me to just change ChakraBytecodeStore::persistBytecode to use a more raw buffer API instead of JSBigString.